src/System.IO.FileSystem/src/System/IO/Directory.cs

```diff
@@ -563,6 +563,9 @@ public static void Move(String sourceDirName, String destDirName)
             String destinationRoot = Path.GetPathRoot(destPath);
             if (!String.Equals(sourceRoot, destinationRoot, pathComparison))
                 throw new IOException(SR.IO_SourceDestMustHaveSameRoot);
+            
+            if (FileSystem.Current.DirectoryExists(fulldestDirName))
+                throw new IOException(SR.Format(SR.IO_AlreadyExists_Name, fulldestDirName));
```

Comment:
> In general, the Windows IO API isn't atomic

If you open a file with Create rather than CreateOrOpen/Open, that's atomic in that it's not going to let you stomp on a file that's created concurrently. That's what I'm referring to here.

Now maybe the worst that could happen if the race condition does occur is that we throw the wrong exception or don't they're one at all. Let's be very sure though that we're not going to lose / delete data.